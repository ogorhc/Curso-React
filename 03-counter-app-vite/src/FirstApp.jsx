import PropTypes from 'prop-types'

export const FirstApp = ({ 
    title, 
    subtitle,
    name }) => {

    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
}

// FirstApp.defaultProps = {
//     name: 'Igor',
//     title: 'No hay título',
//     subtitle: 'No hay subtítulo'
// }