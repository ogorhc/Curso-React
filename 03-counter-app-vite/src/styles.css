
html, body {
    background-color: #212123;
    color: white;
    font-family:  Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
    font-size: 1.3rem;
    margin: 10px;
    min-width: 65px;
    border-radius: 20px;
    cursor: pointer;
}