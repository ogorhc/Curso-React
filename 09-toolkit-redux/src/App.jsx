import reactLogo from './assets/react.svg';
import viteLogo from '/vite.svg';
import './App.css';
import { useDispatch, useSelector } from 'react-redux';
import { decrement, increment, incrementedBy } from './slices/counter';

function App() {
  const { counter } = useSelector((state) => state.counter);
  const dispatch = useDispatch();

  return (
    <>
      <div>
        <a href='https://vitejs.dev' target='_blank'>
          <img src={viteLogo} className='logo' alt='Vite logo' />
        </a>
        <a href='https://react.dev' target='_blank'>
          <img src={reactLogo} className='logo react' alt='React logo' />
        </a>
      </div>
      <h1>Vite + React</h1>
      <p>count is {counter}</p>
      <div>
        <button onClick={() => dispatch(increment())}>Increment</button>
      </div>
      <div>
        <button onClick={() => dispatch(decrement())}>Decrement</button>
      </div>
      <div>
        <button onClick={() => dispatch(incrementedBy(2))}>Increment By: 2</button>
      </div>
    </>
  );
}

export default App;
