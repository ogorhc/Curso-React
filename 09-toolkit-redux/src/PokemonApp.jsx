import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getPokemons } from './store/slices/pokemon';

export const PokemonApp = () => {
  const { isLoading, pokemons = [], page } = useSelector((state) => state.pokemons);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getPokemons());
  }, []);
  return (
    <>
      <h1>Pokemon App</h1>
      <hr />
      <span>Loading: {isLoading ? 'true' : 'false'}</span>
      <ul>
        {pokemons.map((pokemon) => (
          <li key={pokemon.name}>{pokemon.name}</li>
        ))}
      </ul>
      <button disabled={isLoading} onClick={() => dispatch(getPokemons(page - 2))}>
        Back
      </button>
      <button disabled={isLoading} onClick={() => dispatch(getPokemons(page))}>
        Next
      </button>
    </>
  );
};
