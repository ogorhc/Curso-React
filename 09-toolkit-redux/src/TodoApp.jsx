import { useState } from 'react';
import { useGetTodoQuery } from './store/api';

export const TodoApp = () => {
  //   const { data: todos = [], isLoading } = useGetTodosQuery();
  const [todoId, setTodoId] = useState(1);
  const { data: todo, isLoading } = useGetTodoQuery(todoId);
  const previousTodo = () => {
    if (todoId <= 1) return;
    setTodoId(todoId - 1);
  };
  const nextTodo = () => {
    setTodoId(todoId + 1);
  };
  return (
    <>
      <h1>Todos - RTK Query</h1>
      <hr />
      <h4>isLoading: {isLoading ? 'true' : 'false'}</h4>
      <pre>{JSON.stringify(todo, null, 3)}</pre>
      {/* <ul>
        {todos.map((todo) => (
          <li key={todo.id}>
            <strong>{todo.completed ? 'Done ' : 'Pending '}</strong>
            {todo.title}
          </li>
        ))}
      </ul> */}

      <button onClick={previousTodo}>Previous Todo</button>
      <button onClick={nextTodo}>Next Todo</button>
    </>
  );
};
