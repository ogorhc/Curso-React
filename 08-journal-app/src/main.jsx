import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';
import { JournalApp } from './JournalApp';
import { BrowserRouter } from 'react-router-dom';

ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
    <React.StrictMode>
      <JournalApp />
    </React.StrictMode>
  </BrowserRouter>,
);
