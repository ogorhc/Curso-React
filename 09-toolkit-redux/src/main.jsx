import React from 'react';
import ReactDOM from 'react-dom/client';
import { Provider } from 'react-redux';
import './index.css';
import { store } from './store/store.js';
import { TodoApp } from './TodoApp.jsx';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <Provider store={store}>
      <TodoApp />
    </Provider>
  </React.StrictMode>,
);
