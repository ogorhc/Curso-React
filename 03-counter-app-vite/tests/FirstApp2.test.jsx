import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Prueba en <FirstApp/>', () => {
    const title = 'Hola,soy Goku'
    const subTitle = 'Soy un subtitulo'
    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={title} />)
        expect(container).toMatchSnapshot()
    })
    test('Debe de mostrar el mensaje "Hola, soy Goku"', () => {
        render(<FirstApp title={title} />)
        expect(screen.getByText(title)).toBeTruthy()
    })
    
    test('Debe de mostrar el titulo en un h1', () => {
        render(<FirstApp title={title} />)
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)
    })
    
    test('Debe de mostrar el subtitulo enviado por props', () => {
        render(
            <FirstApp 
            title={ title }
            subtitle={ subTitle }
            />
            )
        expect(screen.getAllByText(subTitle).length).toBe(2)
    })
})