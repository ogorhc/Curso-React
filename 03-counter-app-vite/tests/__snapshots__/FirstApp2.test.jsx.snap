// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <FirstApp/> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola,soy Goku
  </h1>
  <p />
  <p />
  <p />
</div>
`;
